pkg/client/client_venconn_test.go

```diff
@@ -334,18 +335,25 @@ func fakeTPP(t testing.TB) (*httptest.Server, *x509.Certificate) {
 //
 //	export KUBEBUILDER_ATTACH_CONTROL_PLANE_OUTPUT=true
 func startEnvtest(t testing.TB) (_ *envtest.Environment, _ *rest.Config, kclient ctrlruntime.WithWatch) {
+	// If KUBEBUILDER_ASSETS isn't set, show a warning to the user.
+	if os.Getenv("KUBEBUILDER_ASSETS") == "" {
+		t.Fatalf("KUBEBUILDER_ASSETS isn't set. You can run this test using `make test`.\n" +
+			"But if you prefer not to use `make`, run these two commands first:\n" +
+			"    make _bin/tools/{kube-apiserver,etcd}\n" +
+			"    export KUBEBUILDER_ASSETS=$PWD/_bin/tools")
+	}
```

Comment:
I was slightly annoyed that running `go test` would always end up with the unhelpful message:

> unable to start control plane itself: failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory